### PR TITLE
Process: Deprecate old methods and properties, implement run()

### PR DESCRIPTION
**Pull request by spevans**
- Mark launchPath and currentDirectoryPath as deprecated,
  replace with executableURL and currentDirectoryURL.

- Implement run() and Process.run(_:arguments:terminationHandler:)
  and mark launch() and Process.launchedProcess(path:arguments:) as
  deprecated.

- Update DarwinCompatibility Tests to 10.13 to use run().

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

File: Foundation/Process.swift [hunk 1]
```diff
@@ -162,48 +162,36 @@ open class Process: NSObject {
...
-                return nil
-            }
-            
-            return URL(fileURLWithPath: launchPath)
-        }
-        set {
-            self.launchPath = newValue?.path
-        }
+
+    @available(*, deprecated: 4, renamed: "executableURL")
```
Comment (millenomi) [inline Foundation/Process.swift]: While this is allowed, I don't think the semantics for this are well-defined. (The '4' in the `deprecated:` portion is meant to be the version of the platform specified just after `@available(`, but in this case there's none.)

The semantically appropriate thing to do here, I think, is just to mark it:

```swift
@available(*, deprecated, renamed:…)
```

Comment (millenomi) [reply Foundation/Process.swift]: Unless it is undocumented that the Swift version is used even if the tag is not `@available(swift, …)`

Comment (spevans) [reply Foundation/Process.swift]: I wasn't sure if a version number was required, its fixed now.

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test and merge

**Pull request merged**